src/main/java/com/google/cloud/tools/sdk/CloudSdkCliExecutor.java

```diff
@@ -0,0 +1,60 @@
+package com.google.cloud.tools.sdk;
+
+import com.google.cloud.tools.internal.process.AsynchronousProcessRunnerRunner;
+import com.google.cloud.tools.internal.process.ProcessRunner;
+import com.google.cloud.tools.internal.process.ProcessRunnerException;
+import com.google.cloud.tools.internal.process.SimpleProcessRunner;
+import com.google.cloud.tools.internal.sdk.CloudSdk;
+import com.google.cloud.tools.internal.sdk.CloudSdkCommandFactory;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.Future;
+
+/**
+ * Created by appu on 3/30/16.
+ */
+public class CloudSdkCliExecutor implements CloudSdkExecutor {
```

Comment:
Well assuming we had other ways of doing cloud sdk calls? Perhaps we don't want to use the cli for doing "modules list" and instead want to use the projects api, we could potentially replace the executor with one that does direct api calls.

Anyway, the executor + action model seems a little wonky and redundant. We might need to define the abstraction differently.